src/components/header/header.spec.js

```diff
@@ -0,0 +1,116 @@
+import { renderComponent, setTestPage } from '../../tests/helpers/rendering';
+import { EXAMPLE_HEADER_SEARCH_AND_MENU_LINKS } from './_test-examples';
+import { getNodeAttributes } from '../../tests/helpers/puppeteer';
+
+describe('script: header', () => {
+    beforeEach(async () => {
+        await setTestPage('/test', renderComponent('header', { ...EXAMPLE_HEADER_SEARCH_AND_MENU_LINKS, variants: 'basic' }));
+    });
+
+    describe('WHEN: the search button is clicked', () => {
+        beforeEach(async () => {
+            await page.click('.ons-js-toggle-header-search');
+        });
+
+        it('THEN: the search input form is displayed', async () => {
```

Comment:
This should be a bit more descriptive of what the test is testing for. That its visible and doesn't have the `ons-u-d-no` class